src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/ActionButtons.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { Col, Row, Button } from 'antd';
+import HandleDownloadManifestClick from './DownloadUtils/HandleDownloadManifestClick';
+import DownloadAllModal from './DownloadAllModal/DownloadAllModal';
+import './ActionButtons.css';
+import DownloadAllFiles from './DownloadUtils/DownloadAllFiles';
+import DownloadJsonFile from './DownloadUtils/DownloadJsonFile';
+
+const ActionButtons = ({ discoveryConfig, resourceInfo }): JSX.Element => {
+  const [downloadStatus, setDownloadStatus] = useState({
+    inProgress: false,
+    message: { title: '', content: <React.Fragment />, active: false },
+  });
+  const studyIDs = [resourceInfo?.study_id];
+
+  return (
+    <div className='discovery-modal_buttons-row' data-testid='actionButtons'>
+      <DownloadAllModal
+        downloadStatus={downloadStatus}
+        setDownloadStatus={setDownloadStatus}
+      />
+      <Row className='row'>
+        {/*
+        <Col flex='1 0 auto'>
+          <Button className='discovery-action-bar-button'>
+            Download <br />
+            Variable-Level Metadata
+          </Button>
+        </Col>
+        */}
+        {discoveryConfig?.features.exportToWorkspace.studyMetadataFieldName
+          && discoveryConfig?.features.exportToWorkspace.enableDownloadStudyMetadata
+          && resourceInfo?.study_metadata && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => DownloadJsonFile(
+                'study-level-metadata',
+                resourceInfo.study_metadata,
+              )}
+            >
+                Download <br />
+                Study-Level Metadata
+            </Button>
+          </Col>
+        )}
+        {discoveryConfig?.features.exportToWorkspace.enableDownloadManifest && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => {
+                HandleDownloadManifestClick(
+                  discoveryConfig,
+                  [resourceInfo],
+                  false,
```

Comment:
This button should behaves exactly the same as its siblings on the Discovery table, that is: if user is not logged in, display button label as `Login to Download Manifest`, and redirects user to the login page on clicking, then redirects user back after loggin in. Right now this button is enabled even for unauthenticated users

Reply:
Created a custom hook for this, UseHandleRedirectToLoginClick. Passed prop isUserLoggedIn from DiscoveryDetails to DataDownloadList and have buttons for each use case now.

Reply:
Wrote a custom hook UseHandleRedirectToLoginClick.ts to redirect the user to login page on clicking if logged out, then to download once logged in. Created unique buttons for each use case, which conditionally render based on new prop for component ```isUserLoggedIn``` determined in DiscoveryDetails